server/src/storage/segment.rs

```diff
@@ -123,6 +125,12 @@ impl UncompressedSegment for SegmentBuilder {
     fn get_memory_size(&self) -> usize {
         SegmentBuilder::get_memory_size()
     }
+
+    /// Spill the in-memory segment to a Parquet file and return Ok when finished.
+    fn spill_to_parquet(&mut self, key: String) -> Result<SpilledSegment, String> {
+        let batch = self.get_record_batch();
+        Ok(SpilledSegment::new(key.clone(), batch))
```

Comment:
Will the many `unwrap()` calls be removed soon or is the `StorageEngine` intentionally designed to `pacnic!()` without attempting recovery when an error occur?

Generally, I do not think `unwrap()` should be used to handle operations that can fail, e.g., `write_batch_to_parquet()` and `SpilledSegment::new()`, unless documented invariants ensure that the error cannot occur. Since writing files to disk can [fail](https://danluu.com/file-consistency/) in [many](https://danluu.com/filesystem-errors/) [interesting](https://danluu.com/deconstruct-files/) [ways](https://wiki.postgresql.org/wiki/Fsync_Errors), I think the `StorageEngine` generally should return errors and allow the caller to decide what to do instead of calling `panic!()` through `unwrap()`.

Reply:
I agree that `unwrap()` should be used a lot less than it is right now. In general there needs to be more strict error handling in the storage engine so I have added attempting to remove `unwrap()` to that "feature".